logstreams/src/main/java/io/zeebe/distributedlog/restore/snapshot/RestoreSnapshotReplicator.java

```diff
@@ -108,11 +104,9 @@ private void restoreInternal(
                 return;
               }
 
-              if (snapshotConsumer.moveValidSnapshot(snapshotId)) {
-                final Long exportedPosition =
-                    restoreContext.getExporterPositionSupplier(partitionId).get();
-                final Long processedPosition =
-                    restoreContext.getProcessorPositionSupplier(partitionId).get();
+              if (snapshotConsumer.completeSnapshot(snapshotId)) {
+                final Long exportedPosition = restoreContext.getExporterPositionSupplier().get();
```

Comment:
Ya. It can be inefficient. I just updated it to open DB once and return a tuple. I also kept the `getMinimiumExportedPosition` because it is used by the deletion service. It doesn't need to read the processed position.